tasks/main.yml

```diff
@@ -40,14 +62,46 @@
     - { name: "docker-py", version: "1.10.5" }
     - { name: "docker-compose", version: "1.9.0" }
 
-- name: start docker
-  service: name=docker state=started enabled=yes
+- name: format docker storage
+  filesystem:
+    dev: "{{ docker_storage_device }}"
+    fstype: xfs
+  when: docker_storage_device is defined
   become: True
 
-- name: load overlay kernel module for docker
-  modprobe:
-    name: overlay
-    state: present
+- name: remove default mount
+  mount:
+    path: /mnt
+    src: /dev/vdb
+    state: absent
+  become: True
+  when: docker_storage_device is defined
+
+- name: mount docker storage
+  mount:
+    path: /var/lib/docker
+    src: "{{ docker_storage_device }}"
+    fstype: xfs
+    opts: noatime
+    state: mounted
+  when: docker_storage_device is defined
+  become: True
+
+- name: setup docker storage
+  blockinfile:
+    dest: /etc/sysconfig/docker-storage-setup
+    block: |
+      {% if docker_storage_driver is defined %}
+      STORAGE_DRIVER={{ docker_storage_driver }}
```

Comment:
What about setting a default for docker_storage_driver here instead of using a conditional? If it was done that way, then configuration of the device could also be split into its own block/file as mentioned in a comment above.

Reply:
To clarify: setting up the /etc/sysconfig/docker-storage-setup file would then be split into two tasks.

Reply:
Not sure if we can come up with a better default than leaving it out and using whatever is the default.